plugins/modules/files/archive.py

```diff
@@ -470,6 +449,20 @@ def result(self):
             'expanded_exclude_paths': [_to_native(p) for p in self.expanded_exclude_paths],
         }
 
+    @property
+    def root(self):
+        return os.path.join(
```

Comment:
Wouldn't it make sense to cache this value, instead of recomputing it every time `self.root` is needed?

Reply:
In the current state yes. If `self.paths` is modified outside of `__init__` in the future, then some extra reasoning will be required. At least `self.root = common_path(self.paths)` makes that a little more obvious so should be fine.